File: Leetcode/Dart/258.dart
class Solution {
  int addDigits(int num) {
    while (num >= 10) {
        int tmp = 0;
        while (num > 0) {
            tmp += num % 10;
            num = num ~/ 10; // 整除
        }
        num = tmp;
    }
    return num;
  }
}

void main(){
  Solution solution = new Solution();
  print(solution.addDigits(38)); // 2
  print(solution.addDigits(0)); // 0
}

File: Leetcode/cpp/easy/258.cpp
/*
 * @lc app=leetcode id=258 lang=cpp
 * @lcpr version=30112
 *
 * [258] Add Digits
 */


// @lcpr-template-start
#include <bits/stdc++.h>
using namespace std;
// @lcpr-template-end
// @lc code=start
class Solution {
public:
    int addDigits(int num) {
        int sod = 0; // sum of digits
        while (num >= 10){
            sod = 0;
            while (num > 0){
                sod += num % 10;
                num /= 10;
            }
            num = sod;
        }
        return num;
    }
};
// @lc code=end



/*
// @lcpr case=start
// 38\n
// @lcpr case=end

// @lcpr case=start
// 0\n
// @lcpr case=end

 */

